src/libAtomVM/bif.c

```diff
@@ -203,71 +225,73 @@ term bif_erlang_is_map_key_2(Context *ctx, term arg1, term arg2)
         case TERM_MAP_NOT_FOUND:
             return FALSE_ATOM;
         case TERM_MAP_MEMORY_ALLOC_FAIL:
-            RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+            RAISE_ERROR_BIF(fail_label, OUT_OF_MEMORY_ATOM);
         default:
             return TRUE_ATOM;
     }
 }
 
-term bif_erlang_length_1(Context *ctx, int live, term arg1)
+term bif_erlang_length_1(Context *ctx, uint32_t fail_label, int live, term arg1)
 {
     UNUSED(live);
 
-    VALIDATE_VALUE(arg1, term_is_list);
+    VALIDATE_VALUE_BIF(fail_label, arg1, term_is_list);
 
     int proper;
     avm_int_t len = term_list_length(arg1, &proper);
     if (UNLIKELY(!proper)) {
-        RAISE_ERROR(BADARG_ATOM);
+        RAISE_ERROR_BIF(fail_label, BADARG_ATOM);
     }
 
     return term_from_int(len);
 }
 
-term bif_erlang_hd_1(Context *ctx, term arg1)
+term bif_erlang_hd_1(Context *ctx, uint32_t fail_label, term arg1)
 {
-    VALIDATE_VALUE(arg1, term_is_nonempty_list);
+    VALIDATE_VALUE_BIF(fail_label, arg1, term_is_nonempty_list);
 
     return term_get_list_head(arg1);
 }
 
-term bif_erlang_tl_1(Context *ctx, term arg1)
+term bif_erlang_tl_1(Context *ctx, uint32_t fail_label, term arg1)
 {
-    VALIDATE_VALUE(arg1, term_is_nonempty_list);
+    VALIDATE_VALUE_BIF(fail_label, arg1, term_is_nonempty_list);
 
     return term_get_list_tail(arg1);
 }
 
-term bif_erlang_element_2(Context *ctx, term arg1, term arg2)
+term bif_erlang_element_2(Context *ctx, uint32_t fail_label, term arg1, term arg2)
 {
-    VALIDATE_VALUE(arg1, term_is_integer);
-    VALIDATE_VALUE(arg2, term_is_tuple);
+    VALIDATE_VALUE_BIF(fail_label, arg1, term_is_integer);
+    VALIDATE_VALUE_BIF(fail_label, arg2, term_is_tuple);
 
     // indexes are 1 based
     int elem_index = term_to_int32(arg1) - 1;
     if (LIKELY((elem_index >= 0) && (elem_index < term_get_tuple_arity(arg2)))) {
         return term_get_tuple_element(arg2, elem_index);
-
     } else {
-        RAISE_ERROR(BADARG_ATOM);
+        RAISE_ERROR_BIF(fail_label, BADARG_ATOM);
     }
 }
 
-term bif_erlang_tuple_size_1(Context *ctx, term arg1)
+term bif_erlang_tuple_size_1(Context *ctx, uint32_t fail_label, term arg1)
 {
-    VALIDATE_VALUE(arg1, term_is_tuple);
+    VALIDATE_VALUE_BIF(fail_label, arg1, term_is_tuple);
 
     return term_from_int32(term_get_tuple_arity(arg1));
 }
 
-term bif_erlang_map_size_1(Context *ctx, int live, term arg1)
+term bif_erlang_map_size_1(Context *ctx, uint32_t fail_label, int live, term arg1)
 {
     UNUSED(live);
 
     if (!UNLIKELY(term_is_map(arg1))) {
+        if (fail_label) {
```

Comment:
Is this a leftover?

Reply:
Nope. We don't want to return immediately if there is no fail_over, we need to build the error term first.